### PR TITLE
feat(content): semi-plausible smokeless gunpowder recipe

### PR DESCRIPTION
**Pull request by YeOldeMiller**
## Summary

SUMMARY: Content "Introduce a semi-plausible smokeless gunpowder recipe"

## Purpose of change

Ah, yes. The holy grail of Cata chemistry. The recipe whose inclusion a certain dev hellbent on realism has been stonewalling for years. That person has no power here though, so here we go.

## Describe the solution

I did some research and tried to come up with a plausible-looking, or at least not entirely random list of inputs, while keeping to existing resources and avoiding introduction of superfluous intermediary products. If we can abstract creation of glycerol and say that lard plus acid makes dynamite, we can certainly make the same leap with gun cotton.

The process requires a proper chemistry set and a pasta extruder, which supposedly can be used for granulation. I also gave it a high skill requirement of 8 to signify the process being dangerous and requiring extreme precision, and a relatively long crafting time, although nowhere near long enough compared to the real thing.  There's no way we can make the process take an actually realistic time using the existing crafting framework, so in the end those variables are mostly about balancing rather than realism. I welcome suggestions on how to adjust them, if necessary.

UPD: added a small amount of ether to components, since apparently you need a solvent to be able to extrude granules. Ether isn't used in a lot of recipes, and in most it's replaceable with a more common alternative, so I feel good about having one more actual use for it.

## Describe alternatives you've considered

Keeping proper gunpowder forever unattainable by post-apocalyptic chemists.

## Testing

Use debug menu to unlock all recipes if necessary. Make sure the new recipe is there.

### CODE REVIEW

**Comment by chaosvolt** [comment]
Seems reasonable, nitrating the hell out of paper or cotton is indeed a good method to base it on. This could in theory justify allowing a given amount of paper or rags as alternatives to cotton balls if you feel it works balance-wise.

In addition, note that the old recipe for it exists in Craftable Gunpowder mod: https://github.com/cataclysmbnteam/Cataclysm-BN/blob/upload/data/mods/Craft_Gunpowder/cgp_recipes.json#L32

Since it's being replaced by a recipe in mainline, it'd be safe to outright delete the mod's version without affecting saves using the mod.

**Comment by chaosvolt** [comment]
> From my research, it appears that wood cellulose doesn't perform as well as cotton IRL for this purpose, and since paper (as well as rags) already has more uses in the game than cotton balls, I didn't feel compelled to provide the alternative. If anything, I welcome the idea of finally having a use for all those pillows.

Fair, being limited to cotton balls should be fine in that case.

> Wouldn't deleting the mod itself break worlds depending on it? Also, the mod provides some extra recipes for a few guns, although I doubt many people use it for that. If none of those items are introduced by the mod, just the recipes, then it should indeed be safe to remove. Should I do that as part of this PR?

No no no, I meant deleting just the version of the recipe that's in the mod, not the entire mod.

**Comment by YeOldeMiller** [comment]
> I meant deleting just the version of the recipe that's in the mod, not the entire mod.

Done. I've also defined `id_suffix` for the recipe in case someone insists on installing the old unaltered version of CGP from Kenan's or elsewhere.

File: data/json/recipes/ammo/components.json [hunk 1]
```diff
@@ -44,6 +44,30 @@
     "tools": [ [ [ "hotplate", 5 ], [ "toolset", 5 ] ] ],
     "components": [ [ [ "oxy_powder", 5 ] ], [ [ "lye_powder", 5 ] ], [ [ "charcoal", 1 ], [ "coal_lump", 1 ] ] ]
   },
+  {
+    "type": "recipe",
+    "result": "gunpowder",
+    "id_suffix": "from_cotton",
```
Comment (chaosvolt) [inline data/json/recipes/ammo/components.json:50]: Adding the `id_suffix` completely breaks the entire "now the base recipe exists in mainline so saves with CGP mod won't error" thing I pointed out.

Comment (YeOldeMiller) [reply data/json/recipes/ammo/components.json:50]: Ah, I see, never used recipe mods so I didn't realize that's how it works. Let me fix this then.

Comment (YeOldeMiller) [reply data/json/recipes/ammo/components.json]: Ok, should be all done now.

**Pull request merged**